### PR TITLE
Fix WPML Views v2 URL translation

### PR DESCRIPTION
**Pull request by lucatume**
Issue: https://moderntribe.atlassian.net/browse/TEC-3608

[Screencast](https://drive.google.com/open?id=1DzSqeqHrgjenEbMEGAEK7TG6DFK0BNS6&authuser=luca%40tri.be&usp=drive_fs)

This PR fixes the translation of Views v2 URLs by WPML by adding more filters to the WPML integration.

### CODE REVIEW

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	 *
+	 * @return string|false Either the request language, e.g. `fr`, or `false` to indicate the language could not be
+	 *                      parsed from the request context or that the request context is the current one and there
+	 *                      is no need to translate URLs.
+	 */
+	protected static function get_request_lang() {
+		if ( ! (
+			function_exists( 'icl_object_id' )
+			&& isset( $_COOKIE['wp-wpml_current_language'] )
+		) ) {
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:51]: :no_entry_sign: Due to server-side caching, server-side based client related logic might not work. We recommend implementing client side logic in JavaScript instead.

<sub>hash: 748c73929e2d4b6d0189702ae9e645f1</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+			return false;
+		}
+
+		$request_lang = tribe_get_request_var( 'wp-wpml_current_language', false );
+
+		if ( $request_lang === ICL_LANGUAGE_CODE ) {
+			return false;
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:58]: :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 83ddb73bee29ec138b7874adab607490</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	 *
+	 * @return array<string,mixed> The View template variables, with the URLs there contained translated, if required.
+	 */
+	public static function translate_template_vars_urls( $template_vars = [] ) {
+		if ( ! is_array( $template_vars ) ) {
+			return $template_vars;
+		}
+
+		$url_keys = array_filter( $template_vars, static function ( $value, $key ) {
+			return str_contains( $key, 'url' ) && (bool) filter_var( $value, FILTER_VALIDATE_URL );
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:79]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: f8b5c10732c45040322a248c6522d36d</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+		}, ARRAY_FILTER_USE_BOTH );
+
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:81]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 72fbbf165cea978485699af4d7cb0c44</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+		}
+
+		$lang = static::get_request_lang();
+
+		if ( false === $lang ) {
+			return $public_views;
+		}
+
+		array_walk( $public_views, static function ( $view_data ) use ( $lang ) {
+			if ( ! isset( $view_data->view_url ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:120]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 2897ba537f5807adc6d2a554ee2e5a1c</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+				return;
+			}
+			$view_data->view_url = apply_filters( 'wpml_permalink', $view_data->view_url, $lang );
+		} );
+
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:125]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 4b5f8d15e63573a96a7abd3819ec5e30</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 9 error(s)

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+	 *
+	 * @return array<string,mixed> The View template variables, with the URLs there contained translated, if required.
+	 */
+	public static function translate_template_vars_urls( $template_vars = [] ) {
+		if ( ! is_array( $template_vars ) ) {
+			return $template_vars;
+		}
+
+		$url_keys = array_filter( $template_vars, static function ( $value, $key ) {
+			return str_contains( $key, 'url' ) && (bool) filter_var( $value, FILTER_VALIDATE_URL );
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:65]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: a839ffeefe5f465c43f205f2b8586679</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+		}, ARRAY_FILTER_USE_BOTH );
+
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:67]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 06b6c702f10c64803435623097aee229</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+		}
+
+		$lang = static::get_request_lang();
+
+		if ( false === $lang ) {
+			return $public_views;
+		}
+
+		array_walk( $public_views, static function ( $view_data ) use ( $lang ) {
+			if ( ! isset( $view_data->view_url ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:106]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: e95ffac29f047089c6a7f906fe54f37b</sub>

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+				return;
+			}
+			$view_data->view_url = apply_filters( 'wpml_permalink', $view_data->view_url, $lang );
+		} );
+
```
Comment (tr1b0t) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:111]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: ed4678a8219253a91b79d14b471152b4</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 7 error(s)

File: src/Tribe/Integrations/WPML/Views/V2/Filters.php [hunk 11]
```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Manages integration between WPML and Views v2 of The Events Calendar.
```
Comment (Camwyn) [inline src/Tribe/Integrations/WPML/Views/V2/Filters.php:3]:
```suggestion
 * Manages integration between WPML and Views v2 filters of The Events Calendar.
```
?

**Review by Camwyn — APPROVED** [review]
One possible docblock tweak, otherwise 🚀

**Review by jesseeproductions — DISMISSED** [review]

**Pull request merged**